### ISSUE TITLE
app.requestSingleInstanceLock doesnt work for signed mas builds (Mac App Store)

### ISSUE THREAD

**Issue by steffanhalv** [electron/electron#15958]
* Output of `node_modules/.bin/electron --version`: 3.0.10
* Operating System (Platform and Version): MacOS Mojave 10.14.1
* Output of `node_modules/.bin/electron --version` on last known working Electron version (if applicable):

**Expected Behavior**
Following code in index.js should not quit app:
```js
let mainInstance = app.requestSingleInstanceLock()
if (!mainInstance) {
  app.quit()
}
```

**Actual behavior**
App quits.

**To Reproduce**
Your best chance of getting this bug looked at quickly is to provide a REPOSITORY that can be cloned and run.

You can fork [electron-quick-start](https://github.com/electron/electron-quick-start) and include a link to the branch with your changes.

If you provide a URL, please list the commands required to clone/setup/run your repo e.g.
```sh
$ git clone $YOUR_URL -b $BRANCH
$ npm install
$ npm start || electron .
```
**Screenshots**

**Additional Information**
This works for unsigned mas builds, but not for signed.

**Comment by yantze** [comment]
Same as this #15304

**Comment by kaansoral** [comment]
I'm affected by this issue too, using `app.requestSingleInstanceLock()` is pretty much a must for macos, and the submitted app fails the mac app store review - sadly the original issue/report didn't pursue a fix

**Comment by kaansoral** [comment]
Ultimately, my solution was to add `<key>LSMultipleInstancesProhibited</key>` to the plist, tested in mas, it worked as expected

**Comment by steffanhalv** [comment]
@kaansoral Thx, so if I add `<key>LSMultipleInstancesProhibited</key>` to plist `app.requestSingleInstanceLock()` will work?

I will test it tomorrow :)

**Comment by kaansoral** [comment]
@steffanhalv No, as far as I gather, it's no trivial task solving the actual issue, couldn't find a way to run the electron app with custom environment variables, this solution just prevents the app from having multiple instances, so you don't have to use `app.requestSingleInstanceLock()` at all - `electron-packager` has an `extend-info` command for easy plist injection by the way

This is my electron-packager issue: https://github.com/electron-userland/electron-packager/issues/923 - I believe it should be possible without such an extreme modification, but didn't pursue the issue, this solution is simpler (electron-builder seems to have some arguments to set env. variables for the first launch, but didn't test or inspect either)

**Comment by steffanhalv** [comment]
Awh I see. I'm using `InstanceLock()` programatically for just a few cases, so block it completely would not be an alternative for my use case :-/ Thx anyway ;) I hope someone from the electron team can fix this ^^

**Comment by dpalou** [comment]
I have the same issue, and I tried using LSMultipleInstancesProhibited to force a single instance, but if I try to run the signed app it crashes with this error:

EXC_CRASH (Code Signature Invalid)

This error doesn't happen without LSMultipleInstancesProhibited, or if I don't sign the app.  I'm using electron-builder to build and sign the app. I also tried using electron-osx-sign to sign the app, with the same result. Any idea of why is this happening?

**Comment by Slapbox** [comment]
I'm having the same issue. `app.requestSingleInstanceLock()` always returns `false` in a `mas-dev` package.

`<key>LSMultipleInstancesProhibited</key>` seems like a promising solution, but I still need `requestSingleInstanceLock` on other platforms and when the app is run via a `dmg`. 

Is there some way to determine that we're running a `MAS` file so we can skip checking for lock? For example, when you're running an `AppImage` you can tell from `process.env.APPIMAGE`

**Comment by vanejung** [comment]
Had the same issue that Apple rejected due to the reason "App quite on launch".

Not the best practice, but used `process.platform` in order to run `app.requestSingleInstanceLock()` only on other platforms but Mac. Which of course fixed the problem, and Apple approved the app.

It wasn't worth the stress and time to make it work for Mac on my end since the OS already doesn't let users run the app more than once. And starting the app via command line is tricky for ordinary users anyway.

But in the long term, hopefully this issue gets fixed so `app.requestSingleInstanceLock()` plays nicely with the MAS build.

**Comment by HaNdTriX** [comment]
@Slapbox you can use 

```js
const isMas = process.mas === true
```

to detect mas builds

**Comment by mahnunchik** [comment]
Any news?

**Comment by mahnunchik** [comment]
Hi @MarshallOfSound @codebytere this issue is from 2018. Do you plan to fix it?

**Comment by mahnunchik** [comment]
Any news from May?

**Comment by idoodler** [comment]
Any news?

**Comment by bojanbizjak** [comment]
When using app.requestSingleInstanceLock(), the application closes immediately after startup.

MacBook Pro (13-inch, M1, 2020)
macOS Big Sur (11.5.2)

"electron": "13.2.2"
"electron-builder": "22.11.7"

"type": "development",
 "hardenedRuntime": false,
 "gatekeeperAssess": false or "gatekeeperAssess": true
